sql/signal_handler.cc

```diff
@@ -223,6 +223,13 @@ extern "C" sig_handler handle_fatal_signal(int sig)
     my_safe_printf_stderr("\nConnection ID (thread ID): %lu\n",
                           (ulong) thd->thread_id);
     my_safe_printf_stderr("Status: %s\n\n", kreason);
+
+    StringBuffer<128> buf;
+    if (!print_explain_for_slow_log(thd->lex, thd, &buf))
+    {
+      my_safe_printf_stderr("%s\n",  buf.c_ptr_safe());
+    }
```

Comment:
I'd say it is horribly complex for signal handler. But I have no strong preference whether to have it or not. I'd let somebody else to decide.

Reply:
Its a bit complex, however given how much pain @elenst goes though trying to get this info accurately into JIRA issues I thought its worth it.

Reply:
Is my_safe_printf_stderr non-reentrant ? In other words, it is possible that function will get interrupted before it finishes and get called again as part of another signal handler ?

Reply:
my_safe_printf_stderr definitely is as its used elsewhere in the signal handler.

print_explain_for_slow_log seems to just grap information from the THD and push it to the buffer. I couldn't see any wait/local conditions associated with it.